### ISSUE TITLE
Audio data duplicates when pasted into another project

### ISSUE THREAD

**Issue by vsverchinsky** [audacity/audacity#4047]
### Bug description

 Pasting audio into another project may create redundant copies, increasing resulting project's size without necessity.

### Steps to reproduce

1. Create empty project and generate long (e.g. 1hr) mono audio track
2. Select whole generated clip (Ctrl+A) and copy to clipboard (Ctrl+C)
3. Create new project (Ctrl+N) and add two or more **empty** mono audio tracks
4. Select all empty tracks in the second project and paste from clipboard (Ctrl+V)
5. Save both projects and compare resulting sizes

### Expected behavior

Audio data uniquely copied to the second project and referenced from each track where it get pasted, as it happens when audio is copied within a single project.

### Actual behavior

Each track gets it's own copy of the original audio data, increasing project's size.

### Audacity Version

current master / alpha / nightly build

### Operating system

Windows, macOS

**Comment by chuckkh** [comment]
I can't get the two projects to have similar sizes no matter what I do. I paste 1 hour of silence on *one* track and the size jumps to 600 MB.

**Comment by virgilON** [comment]
This is the #3280 issue, which has a pending solution. 
I've suggested a further enhancement to deal with the havoc left behind by this issue: a 'purge' function that would permanently remove the hidden data ( File>Save Project>Save Without SmartClips Data).  While the pending fix is most welcome, I would like to see this additional improvement included in the next version or two at least, in order to allow affected users to easily clean up bloated projects created by recent versions of Audacity.  As a permanent feature, this additional functionality would also rescue future projects that might become inadvertently bloated by unwary users choosing to retain SmartClip functionality when copy/pasting from large projects.
For a kludgy repair for bloated projects, see my suggestion in #3963.

**Comment by crsib** [comment]
Copying within the project does not copy data **at all**. Applying effects from the menu, however, generates new data, with the old data held by the Undo/Redo stack. It is very easy to get the project to enormous size by applying the effects and having a large project history.

What is more important, Audacity only reclaims the disk space when the project is closed. There was a dialog that allowed to compact the project on demand, but it was disabled. I think that @petersampsonaudacity better knows why.

If it is possible, can you please send us an example project for analysis?

**Comment by petersampsonaudacity** [comment]
> What is more important, Audacity only reclaims the disk space when the project is closed. There was a dialog that allowed to compact the project on demand, but it was disabled. I think that @petersampsonaudacity better knows why.

@crsib  Dmitry, we had a number of issues with compaction which led to the on-demand command being withdrawn
https://bugzilla.audacityteam.org/show_bug.cgi?id=2531
https://bugzilla.audacityteam.org/show_bug.cgi?id=2794

and in particular this is the one that eventually led to it being disabled
https://bugzilla.audacityteam.org/show_bug.cgi?id=2600

_Personally I would _**much prefer**_ to see it being brought back if possible._

**Comment by petersampsonaudacity** [comment]
@LWinterberg 

Is this not fixed by the fix for #3820 ?
**Performance and file size issues when copying and pasting a small portion of a very large clip**

**Comment by crsib** [comment]
I don't think so. This issue covers corner case when Audacity will duplicate data under certain conditions, instead of just duplicating references to it.
